pandas/core/dtypes/cast.py

```diff
@@ -133,7 +134,7 @@ def is_nested_object(obj) -> bool:
     return False
 
 
-def maybe_downcast_to_dtype(result, dtype: Dtype):
+def maybe_downcast_to_dtype(result, dtype: Union[str, np.dtype]):
```

Comment:
this is not correct, DtypeObj right

Reply:
has to be a numpy type or str according to `mypy`. When I put it `DtypeObj` `mypy` errors:
```
pandas/core/dtypes/cast.py:180: error: Item "ExtensionDtype" of "Union[Any, ExtensionDtype]" has no attribute "freq"  [union-attr]
```